src/org/labkey/test/tests/list/ListTest.java

```diff
@@ -683,6 +689,91 @@ public void testBatchInsertErrors() throws IOException, CommandException
+        );
+        String listName = "remoteApiListTestAddColumn";
+        FieldDefinition.LookupInfo info = new FieldDefinition.LookupInfo(getProjectName(), "lists", listName);
+        TestDataGenerator dgen = new TestDataGenerator(info)
+                .withColumns(cols);
+        DomainResponse createResponse = dgen.createList(createDefaultConnection(true), "key");
+        Domain listDomain = createResponse.getDomain();
+        List<PropertyDescriptor> listFields = createResponse.getDomain().getFields();
+        listFields.add(new FieldDefinition("volume", FieldDefinition.ColumnType.Decimal));
+        listDomain.setFields(listFields);
+
+        // now save with an extra field
+        SaveDomainCommand saveCmd = new SaveDomainCommand(info.getSchema(), info.getTable());
+        saveCmd.setDomainDesign(listDomain);
+        DomainResponse saveResponse = saveCmd.execute(createDefaultConnection(true), info.getFolder());
+
+        // now verify
+        assertEquals(listFields.size(), saveResponse.getDomain().getFields().size());
+        for (PropertyDescriptor expectedField : listFields)
+        {
+            checker().verifyTrue( "expect field [" + expectedField.getName() + "] with type [" +expectedField.getRangeURI()+ "]",
+                    saveResponse.getDomain().getFields().stream()
+                    .anyMatch(a -> a.getName().equals(expectedField.getName()) &&
+                            a.getRangeURI().endsWith(expectedField.getRangeURI())));
+        }
+    }
+
+    @Test
+    public void testRemoveColumnOverAPI() throws Exception
+    {
+        List<FieldDefinition> cols = Arrays.asList(
+                new FieldDefinition("name", FieldDefinition.ColumnType.String),
+                new FieldDefinition("title", FieldDefinition.ColumnType.String),
+                new FieldDefinition("dewey", FieldDefinition.ColumnType.Decimal),
+                new FieldDefinition("removeMe", FieldDefinition.ColumnType.Decimal)
+        );
+        String listName = "remoteApiListTestRemoveColumn";
+        FieldDefinition.LookupInfo info = new FieldDefinition.LookupInfo(getProjectName(), "lists", listName);
+        TestDataGenerator dgen = new TestDataGenerator(info)
+                .withColumns(cols);
+        DomainResponse createResponse = dgen.createList(createDefaultConnection(true), "key");
+        Domain listDomain = createResponse.getDomain();
+        List<PropertyDescriptor> listFields = createResponse.getDomain().getFields();
+        listFields.removeIf(a-> a.getName().equals("removeMe"));
+        listDomain.setFields(listFields);
+
+        SaveDomainCommand saveCmd = new SaveDomainCommand(info.getSchema(), info.getTable());
+        saveCmd.setDomainDesign(listDomain);
+        DomainResponse saveResponse = saveCmd.execute(createDefaultConnection(true), info.getFolder());
+
+        checker().verifyFalse("expect field [title] with type [" +FieldDefinition.ColumnType.String.getConceptURI()+ "]",
```

Comment:
What is `title`? Copy/paste error?
```suggestion
        checker().verifyFalse("'removeMe' field was not deleted.",
```